lib/msf/core/post/file.rb

```diff
@@ -539,6 +551,8 @@ def rename_file(old_file, new_file)
   def copy_file(src_file, dst_file)
     if session.type == "meterpreter"
       return (session.fs.file.cp(src_file, dst_file).result == 0)
+    elsif session.type == 'powershell'
+      cmd_exec("Copy-Item \"#{src_file}\" -Destination \"#{dst_file}\"")
```

Comment:
If the `dst_file` path is a directory, Meterpreter will raise an exception while the Powershell one will place the file in that directory, keeping it's name. Both should be updated to consistently handle what happens when the dst_file parameter is a directory.

Should also test and compare what happens when `src_file` is a directory.

Reply:
okay sure I'll do that :+1: